zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -486,16 +493,54 @@ mod tests {
         Ok(())
     }
 
-    fn arbitrary_chain(tip_height: block::Height) -> BoxedStrategy<Vec<Arc<Block>>> {
-        Block::partial_chain_strategy(LedgerState::new(tip_height, Network::Mainnet), 100)
+    #[derive(Debug)]
+    struct PreparedChainTree {
+        chain: Arc<Vec<PreparedBlock>>,
+        count: BinarySearch,
     }
 
-    prop_compose! {
-        fn arbitrary_chain_and_count()
-            (chain in arbitrary_chain(NetworkUpgrade::Blossom.activation_height(Network::Mainnet).unwrap()))
-            (count in 1..chain.len(), chain in Just(chain)) -> (SummaryDebug<Vec<Arc<Block>>>, usize)
-        {
-            (SummaryDebug(chain), count)
+    impl ValueTree for PreparedChainTree {
+        type Value = (Arc<Vec<PreparedBlock>>, <BinarySearch as ValueTree>::Value);
+
+        fn current(&self) -> Self::Value {
+            (self.chain.clone(), self.count.current())
+        }
+
+        fn simplify(&mut self) -> bool {
+            self.count.simplify()
+        }
+
+        fn complicate(&mut self) -> bool {
+            self.count.complicate()
+        }
+    }
+
+    #[derive(Debug, Default)]
+    struct PreparedChain {
+        chain: Mutex<Option<Arc<Vec<PreparedBlock>>>>,
```

Comment:
We're about to replace all our sync::Mutexes with futures-aware Mutexes, but we'll want to keep this one (#1995)
 
```suggestion
        // the proptests are threaded (not async), so we want to use a threaded mutex here
        chain: std::sync::Mutex<Option<Arc<Vec<PreparedBlock>>>>,
```